packages/babel-parser/src/plugins/flow/index.js

```diff
@@ -1619,24 +1619,38 @@ export default (superClass: Class<Parser>): Class<Parser> =>
     }
 
     flowParsePostfixType(): N.FlowTypeAnnotation {
-      const startPos = this.state.start,
-        startLoc = this.state.startLoc;
+      const startPos = this.state.start;
+      const startLoc = this.state.startLoc;
       let type = this.flowParsePrimaryType();
-      while (this.match(tt.bracketL) && !this.canInsertSemicolon()) {
+      let seenOptionalIndexedAccess = false;
+      while (
+        (this.match(tt.bracketL) || this.match(tt.questionDot)) &&
+        !this.canInsertSemicolon()
+      ) {
         const node = this.startNodeAt(startPos, startLoc);
+        const optional = this.eat(tt.questionDot);
+        seenOptionalIndexedAccess ||= optional;
```

Comment:
Can you update the `flow-bin` in `package.json`? Good to know Flow supports logical assignment.

Reply:
Oops, it doesn't I was getting ahead of myself. I'll update this.